src/test/java/com/artipie/rpm/pkg/ModifiableMetadataTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.pkg;
+
+import com.artipie.rpm.Digest;
+import com.artipie.rpm.StandardNamingPolicy;
+import com.artipie.rpm.meta.XmlRepomd;
+import com.jcabi.aspects.Tv;
+import com.jcabi.log.Logger;
+import com.jcabi.matchers.XhtmlMatchers;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.channels.FileChannel;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.logging.Level;
+import javax.xml.stream.XMLStreamException;
+import org.cactoos.list.ListOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsNot;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.redline_rpm.ReadableChannelWrapper;
+import org.redline_rpm.Scanner;
+import org.redline_rpm.header.Format;
+import org.redline_rpm.header.Header;
+
+/**
+ * Test for {@link ModifiableMetadata}.
+ * @since 0.9
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+class ModifiableMetadataTest {
+
+    @Test
+    void generatesMetadataFile(@TempDir final Path temp) throws IOException {
+        final Path res = temp.resolve("primary.xml");
+        res.toFile().createNewFile();
+        final ModifiableMetadata mtd = new ModifiableMetadata(
+            new MetadataFile("primary", new PrimaryOutput(res).start()),
+            Paths.get("src/test/resources-binary/repodata/primary.xml.example")
+        );
+        final Path rpm =
+            Paths.get("src/test/resources-binary/abc-1.01-26.git20200127.fc32.ppc64le.rpm");
+        mtd.accept(new FilePackage.Headers(this.header(rpm), rpm, Digest.SHA256));
+        mtd.close();
```

Comment:
@olenagerasimova we can put `ModifiableMetadata` in a try-with-resources block instead of invoking `close()`, it's a `Closeable` (via `Metadata` and `PackageOutput` interfaces).

Reply:
@carlosmiranda I do not think it's convenient here - after closing `mtd` i call `brush` method

Reply:
@olenagerasimova all right, makes sense